types/defines/queues.d.ts

```diff
@@ -44,4 +44,11 @@ interface Queue<Body = any> {
    * @returns A promise that resolves when the message is confirmed to be written to disk.
    */
   send(message: Body): Promise<void>;
+
+  /**
+   * Sends a batch of messages to the Queue.
+   * @param messages Each item in the array must be supported by the [structured clone algorithm](https://developer.mozilla.org/en-US/docs/Web/API/Web_Workers_API/Structured_clone_algorithm#supported_types). You can send up to 100 messages at a time, though items are limited to 128 KB each, and the total size of the array cannot exceed 256 KB.
+   * @returns A promise that resolves when the messages are confirmed to be written to disk.
+   */
+  sendBatch(messages: Body[]): Promise<void>;
```

Comment:
This is not the correct parameter type.

1. You can't just pass an `sendBatch` an array of anything -- it has to be an array of objects containing a `body` field, with your messages contents contained in that `body` field.
2. Technically `sendBatch` accepts any iterable type, not just an array.

This was the original typescript spec that Ashcon put together a while back:

```
type MessageSendOptions = {
    // Reserved for future things
};

type MessageSendRequest<Body = unknown> = MessageSendOptions & {
    body: Body
};

interface Queue<Body = unknown> {
    send(message: Body, options?: MessageSendOptions): Promise<void>;
    sendBatch(batch: Iterable<MessageSendRequest<Body>>): Promise<void>;
}
```

Reply:
Thanks!